diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -0,0 +1,80 @@
+import com.pinterest.ktlint.core.ast.ElementType.IMPORT_LIST
+import com.pinterest.ktlint.core.ast.ElementType.KDOC_MARKDOWN_INLINE_LINK
+import com.pinterest.ktlint.core.ast.ElementType.KDOC_TEXT
+import com.pinterest.ktlint.core.ast.ElementType.PACKAGE_DIRECTIVE
+import org.cqfn.diktat.common.config.rules.RuleConfiguration
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.common.config.rules.getRuleConfig
+import org.cqfn.diktat.ruleset.constants.Warnings.LONG_LINE
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import java.io.IOException
+import java.net.URL
+
+@Suppress("ForbiddenComment")
+class LineLength : Rule("line-length") {
+
+    companion object {
+        private const val MAX_LENGTH = 120L
+    }
+
+    private lateinit var configRules: List<RulesConfig>
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var fileName: String? = null
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       params: KtLint.Params,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        configRules = params.getDiktatConfigRules()
+        fileName = params.fileName
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        val configuration = LineLengthConfiguration(
+                configRules.getRuleConfig(LONG_LINE)?.configuration ?: mapOf())
+        if (node.elementType == FILE) {
+            node.getChildren(null).forEach {
+                if (it.elementType != PACKAGE_DIRECTIVE || it.elementType != IMPORT_LIST)
+                    checkLength(it, configuration)
+            }
+        }
+    }
+
+    private fun checkLength(node: ASTNode, configuration: LineLengthConfiguration) {
+        var offset = 0
+        node.text.lines().forEach {
+            if (it.length > configuration.lineLength) {
+                val newNode = node.psi.findElementAt(offset + it.trim().length - 1)!!.node
+                if ((newNode.elementType != KDOC_TEXT && newNode.elementType != KDOC_MARKDOWN_INLINE_LINK) || !isKDocValid(newNode)) {
+                    LONG_LINE.warnAndFix(configRules, emitWarn, isFixMode,
+                            "max line length ${configuration.lineLength}, but was ${it.length}",
+                            offset.plus(node.startOffset)) {}
+                }
+            }
+            offset += it.length + 1
+        }
+    }
+
+    // fixme json method
+    private fun isKDocValid(node: ASTNode): Boolean {
+        return try {
+            if (node.elementType == KDOC_TEXT)
+                URL(node.text.split("\\s".toRegex()).last { it.isNotEmpty() }).toURI()
+            else
+                URL(node.text.substring(node.text.indexOfFirst { it == ']' } + 2, node.textLength - 1)).toURI()
+            true
+        } catch (e: IOException) {
```

Comment:
When you are catching `IOException`, you catch only `MalformedURLException`, which is thrown from `URL` constructor. `toUri()` throws `URISyntaxException` which extends `Exception`. I think you could remove `toUri` call and check that calling constructor is enough to cause and exception. Also, you could narrow a catch clause.